File: Prototypes/MATLAB/Sandbox/GSpline/scaleAwayParameters.m
function [scaledParams,throwndata] = scaleAwayParameters(unscaledParams,datapoints)


quad_ids = unique(unscaledParams(:,1));
% 
% scaledParams = zeros(size(unscaledParameters));

scaledParams = unscaledParams(unscaledParams(:,3) >=0 & unscaledParams(:,2) >=0 & unscaledParams(:,2) <= 1 & unscaledParams(:,3) <= 1,:);
throwndata = datapoints(unscaledParams(:,3) >=0 & unscaledParams(:,2) >=0 & unscaledParams(:,2) <= 1 & unscaledParams(:,3) <= 1,:);



for i = 1:length(quad_ids)
    umax = max(scaledParams(scaledParams(:,1) == quad_ids(i),2));
    umin = min(scaledParams(scaledParams(:,1) == quad_ids(i),2));
    vmax = max(scaledParams(scaledParams(:,1) == quad_ids(i),3));
    vmin = min(scaledParams(scaledParams(:,1) == quad_ids(i),3));
    
    uscale = umax - umin;
    vscale = vmax - vmin;
    if(uscale)
        scaledParams(scaledParams(:,1) == quad_ids(i),2) = (scaledParams(scaledParams(:,1) == quad_ids(i),2) - umin)/uscale;
    end
    if(vscale)
        scaledParams(scaledParams(:,1) == quad_ids(i),3) = (scaledParams(scaledParams(:,1) == quad_ids(i),3) - vmin)/vscale;
    end
    
    
end




end

File: Prototypes/PYTHON/Sandbox/GSpline/scaleAwayParameters.py
import numpy as np
import math
import scipy.io as sio


def scaleAwayParameters(unscaled_params, datapoints):
    '''
    :param unscaled_params:
    :param datapoints:
    :return:
    '''

    quad_ids = np.unique(unscaled_params[:, 0]);
    scaledParams = unscaled_params[
                   (unscaled_params[:, 2] >= 0) * (unscaled_params[:, 1] >= 0) * (unscaled_params[:, 1] <=1) * (
                   unscaled_params[:, 2] <= 1), :]
    throwndata = datapoints[(unscaled_params[:, 2] >= 0)* (unscaled_params[:, 1] >= 0)*(
    unscaled_params[:, 1] <= 1)* (unscaled_params[:, 2] <= 1), :];

    for i in range(len(quad_ids)):
        umax = np.max(scaledParams[(scaledParams[:,0] == quad_ids[i]), 1]);
        umin = np.min(scaledParams[(scaledParams[:,0] == quad_ids[i]), 1]);
        vmax = np.max(scaledParams[(scaledParams[:,0] == quad_ids[i]), 2]);
        vmin = np.min(scaledParams[(scaledParams[:,0] == quad_ids[i]), 2]);

        uscale = umax - umin;
        vscale = vmax - vmin;

        if (uscale):
            scaledParams[(scaledParams[:, 0] == quad_ids[i]), 1] = (scaledParams[(
                scaledParams[:, 0] == quad_ids[i]), 1] - umin) / uscale;

        if (vscale):
            scaledParams[(scaledParams[:, 0] == quad_ids[i]), 2] = (scaledParams[(
                scaledParams[:, 0] == quad_ids[i]), 2] - vmin) / vscale;

    return scaledParams, throwndata;


#####MAIN CODE######
# Initialization
parameters = np.genfromtxt('Data/Cantilever_try/parameters.csv', delimiter=';')
quads = np.array(np.genfromtxt('Data/Cantilever_try/cantilever_quads_coarse.csv', delimiter=';'))
vertices = np.array(np.genfromtxt('Data/Cantilever_try/cantilever_verts_coarse.csv', delimiter=';'))
fine_vertices = np.array(np.genfromtxt('Data/Cantilever_try/cantilever_verts_fine.csv', delimiter=';'))


mat_contents = sio.loadmat('Data/Cantilever_try/cantilever.mat')

A = np.array(mat_contents["A"])
B1 = np.array(mat_contents["B1"])
B2 = np.array(mat_contents["B2"])
C = np.array(mat_contents["C"])

print scaleAwayParameters(parameters, fine_vertices)
